stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -1385,6 +1418,59 @@ impl NakamotoChainState {
         Ok(true)
     }
 
+    /// Get the aggregate public key for the given block.
+    pub fn get_aggregate_public_key(
+        sortdb: &SortitionDB,
+        sort_handle: &SortitionHandleConn,
+        chainstate: &mut StacksChainState,
+        header: &NakamotoBlockHeader,
+        canonical_block_header: &StacksHeaderInfo,
```

Comment:
`canonical_block_header` is the block that the `aggregate_public_key` will be fetched at, which isn't necessary the "canonical chain tip" -- can you rename this variable to something like `fetch_at_block`?

Reply:
I think that, specifically, this needs to be at least as high as the last block in the last reward cycle.